File: src/js/src/scopes/HeaderScope.js
function HeaderScope(depth) {
	this.depth = depth
	this.type = ScopeTypeCache.header;
	this.level = ScopeLevel.block;
	this.canNestText = true;
	
	this.open = function(analyzer) {
		analyzer.openHeader(this.depth);
	};
	this.close = function(analyzer) {
		analyzer.closeHeader(this.depth);
	};
}

extend(Scope, HeaderScope);

File: src/dotnet/Src/ButterflyNet.Parser/Scopes/HeaderScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class HeaderScope : BlockScope {
		public HeaderScope(int depth) {
			Depth = depth;
		}

		public int Depth { get; private set; }

		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenHeader(Depth);
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseHeader(Depth);
		}
	}
}